src/components/plots/FrequencyPlot.jsx

```diff
@@ -24,13 +25,17 @@ const FrequencyPlot = (props) => {
     if (!hierarchy && !properties) {
       dispatch(loadCellSets(experimentId));
     }
-  }, [experimentId]);
+  }, []);
 
-  const spec = generateSpec(config, generateData(hierarchy, properties, config));
+  useEffect(() => {
+    if (hierarchy && properties) {
+      setPlotSpec(generateSpec(config, generateData(hierarchy, properties, config)));
+    }
+  }, [hierarchy, properties]);
```

Comment:
This was a major bottleneck, `spec` was generated every time the function rerendered, and since the internal `generateData` function is quite messy and not efficient, it would do a bunch of tasks for no reason, especially on larger data sets.